pygbm/multiclass_notes

```diff
@@ -0,0 +1,45 @@
+Notes about multiclass classification
+=====================================
+
+Adding support for multiclass classification involves a few changes. The
+main one is that instead of building 1 tree per iteration (like in binary
+classification and regression), we build K trees per iteration, where K is
+the number of classes.
+
+Each tree is a kind of OVR tree, but trees are not completely independent
+because they influence each others when the gradients and hessians are
+updated in MultinomialLogistic.update_gradients_and_hessians(). Concretely,
+the K trees of the ith iteration do not depend on each other, but each tree
+at iteration i depends on *all* the K trees of iteration i - 1.
+
+For a given sample, the probability that it belongs to class k is computed
+as a regular softmax between scores = [scores_0, scores_1, ... scores_K-1]
+where scores_k = sum(<leaf value of kth tree of iteration i>
+                     for i in range(n_iterations)).
+The predicted class is then the argmax of the K probabilities.
+
+Regarding implementation details, the arrays
+- y_pred
+- gradients
+- hessians (for non-constant hessians)
+- decision, as returned by _decision_function()
```

Comment:
This doc should be updated to stop mentionning `y_pred` and `_decision_function` and instead use `raw_predictions` and `_raw_predict`.